go/integration_test/interpreter/integration_test.go

```diff
@@ -1109,3 +1109,29 @@ func newMockStateDBForIntegrationTests(ctrl *gomock.Controller) *MockStateDB {
 
 	return mockStateDB
 }
+
+func TestEVM_CanSuccessfullyProcessPcBiggerThanCodeLength(t *testing.T) {
+	for _, variant := range Variants {
+		revision := Istanbul
+		t.Run(fmt.Sprintf("%s/%s", variant, revision), func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			mockStateDB := newMockStateDBForIntegrationTests(ctrl)
+			// all implementations should be able to handle PC that goes beyond the code length.
+			// this is reproducible by not calling STOP, RETURN, REVERT or SELFDESTRUCT at the end of the code.
+			codeReturn := []byte{
+				byte(vm.PUSH1), byte(32),
+			}
+			setDefaultCallStateDBMock(mockStateDB, tosca.Address{}, codeReturn)
+			evm := GetCleanEVM(revision, variant, mockStateDB)
+
+			// Run an interpreter
+			result, err := evm.Run(codeReturn, []byte{})
+
+			// Check the result.
+			if err != nil || !result.Success {
+				t.Errorf("execution should fail, error is: %v, result %v", err, result)
```

Comment:
This error message is going to be very confusing when the tests fails. Although the execution should fail, the error should still be nil. Maybe specify that the execution was not successful and not print the whole result.